tycho-surefire/tycho-surefire-plugin/pom.xml

```diff
@@ -44,6 +44,36 @@
 				<artifactId>maven-surefire-plugin</artifactId>
 				<version>${surefire-version}</version>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<version>3.0.0-M3</version>
+				<executions>
+					<execution>
+						<id>enforce-runtime-jdk-compatibility</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<phase>verify</phase>
+						<configuration>
+							<rules>
+								<enforceBytecodeVersion>
+									<!-- Surefire Booter Eclipse Application and JUnit/TestNG Runners -->
+									<includes>org.eclipse.tycho:org.eclipse.tycho.surefire.*</includes>
```

Comment:
Relying on names is a bit awkward here. Ideally, one would place the `<enforceBytecodeVersion>` rule in the `org.eclipse.tycho.surefire.*` themselves, but they don’t have an _real_ `<dependencies>`, which the `<enforceBytecodeVersion>` rule can check but only `<artifactItems>` for the `dependency:unpack` goal, which it cannot see. So placing the sanity check in `tycho-surefire-plugin` is the best I could come up with.